pkg/controllers/etcd.go

```diff
@@ -36,25 +37,119 @@ var (
 	HealthCheckWait    = time.Duration(3 * time.Second)
 )
 
-type EtcdService struct{}
+var (
+	tlsCipherSuites = []string{
+		"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+		"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+		"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+		"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+		"TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
+		"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
+	}
+)
+
+type EtcdService struct {
+	NodeName            string
+	DataDir             string
+	CertFile            string
+	KeyFile             string
+	TrustedCAFile       string
+	InitialUrl          string
+	PeerCertFile        string
+	PeerKeyFile         string
+	PeerTrustedCAFile   string
+	AdvertiseClientUrls string
+	ListenClientUrls    string
+	ListenPeerUrls      string
+	ListenMetricsUrls   string
+}
 
 func NewEtcd(cfg *config.MicroshiftConfig) *EtcdService {
-	return &EtcdService{}
+	s := &EtcdService{}
+	s.configure(cfg)
+	return s
+}
+
+func (s *EtcdService) configure(cfg *config.MicroshiftConfig) {
+	s.NodeName = cfg.NodeName
+
+	s.DataDir = filepath.Join(microshiftDataDir, s.Name())
+	certsDir := cryptomaterial.CertsDirectory(microshiftDataDir)
+	etcdServingCertDir := cryptomaterial.EtcdServingCertDir(certsDir)
+	etcdPeerCertDir := cryptomaterial.EtcdPeerCertDir(certsDir)
+	etcdSignerCertPath := cryptomaterial.CACertPath(cryptomaterial.EtcdSignerDir(certsDir))
+
+	s.CertFile = cryptomaterial.PeerCertPath(etcdServingCertDir)
+	s.KeyFile = cryptomaterial.PeerKeyPath(etcdServingCertDir)
+	s.TrustedCAFile = etcdSignerCertPath
+
+	s.PeerCertFile = cryptomaterial.PeerCertPath(etcdPeerCertDir)
+	s.PeerKeyFile = cryptomaterial.PeerKeyPath(etcdPeerCertDir)
+	s.PeerTrustedCAFile = etcdSignerCertPath
+
+	s.InitialUrl = fmt.Sprintf("https://%s:2380", cfg.NodeIP)
+	s.AdvertiseClientUrls = fmt.Sprintf("https://%v:2379", cfg.NodeIP)
+	s.ListenClientUrls = fmt.Sprintf("https://127.0.0.1:2381,https://%v:2381", cfg.NodeIP)
+	s.ListenPeerUrls = fmt.Sprintf("https://%v:2380", cfg.NodeIP)
+	s.ListenMetricsUrls = "127.0.0.1:2381"
 }
 
 func (s *EtcdService) Name() string           { return "etcd" }
 func (s *EtcdService) Dependencies() []string { return []string{} }
 
 func (s *EtcdService) Run(ctx context.Context, ready chan<- struct{}, stopped chan<- struct{}) error {
+	defer close(stopped)
+
+	// Check to see if we should run as a systemd run or directly as a binary.
+	runningAsSvc := os.Getenv("INVOCATION_ID") != ""
+
+	// Get the path to the etcd binary based on the MicroShift binary location.
 	microshiftExecPath, err := os.Executable()
 	if err != nil {
 		return fmt.Errorf("%v failed to get exec path: %v", s.Name(), err)
 	}
 	etcdPath := filepath.Join(filepath.Dir(microshiftExecPath), "microshift-etcd")
-	args := []string{"run"}
 
+	// All of the etcd parameters
+	args := []string{
```

Comment:
Take a look at `microshift/etcd/cmd/microshift-etcd/run.go` - we're not running `etcd` binary directly so we don't need all these arguments

Reply:
Ahh missed that, thanks.